soc/nxp/imxrt/imxrt7xx/cm33/soc.c

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief System/hardware module for NXP RT7XX platform
+ *
+ * This module provides routines to initialize and support board-level
+ * hardware for the RT7XX platforms.
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/linker/sections.h>
+#include <soc.h>
+#include "fsl_power.h"
+#include "fsl_clock.h"
+
+#ifdef CONFIG_HAS_MCUX_CACHE
+#include <fsl_cache.h>
+#endif
+
+#ifdef CONFIG_FLASH_MCUX_XSPI_XIP
+#include "flash_clock_setup.h"
+#endif
+
+#define SYSOSC_SETTLING_US 220U      /*!< System oscillator settling time in us */
+#define XTAL_SYS_CLK_HZ    24000000U /*!< xtal frequency in Hz */
+
+const clock_main_pll_config_t g_mainPllConfig_clock_init = {
+	.main_pll_src = kCLOCK_MainPllOscClk, /* OSC clock */
+	.numerator = 0,   /* Numerator of the SYSPLL0 fractional loop divider is 0 */
+	.denominator = 1, /* Denominator of the SYSPLL0 fractional loop divider is 1 */
+	.main_pll_mult = kCLOCK_MainPllMult22 /* Divide by 22 */
+};
+
+const clock_audio_pll_config_t g_audioPllConfig_clock_init = {
+	.audio_pll_src = kCLOCK_AudioPllOscClk, /* OSC clock */
+	.numerator = 5040,    /* Numerator of the Audio PLL fractional loop divider is 0 */
+	.denominator = 27000, /* Denominator of the Audio PLL fractional loop divider is 1 */
+	.audio_pll_mult = kCLOCK_AudioPllMult22, /* Divide by 22 */
+	.enableVcoOut = true};
+
+void soc_reset_hook(void)
+{
+	SystemInit();
+}
+
+void soc_early_init_hook(void)
```

Comment:
Most of this clock code configured in `soc_early_init_hook()` is board/application specific.  If users want to modify these clocks to match their board, it is not easily done.  Please move all this clock init code to `board_early_init_hook()` in board.c.  Then users can easily modify with their custom board.  Thanks